rpc/client/http/http.go

```diff
@@ -471,24 +471,58 @@ func (c *baseRPCClient) TxSearch(
 	page,
 	perPage *int,
 	orderBy string,
-) (
-	*ctypes.ResultTxSearch, error) {
+) (*ctypes.ResultTxSearch, error) {
+
 	result := new(ctypes.ResultTxSearch)
 	params := map[string]interface{}{
 		"query":    query,
 		"prove":    prove,
 		"order_by": orderBy,
 	}
+
 	if page != nil {
 		params["page"] = page
 	}
 	if perPage != nil {
 		params["per_page"] = perPage
 	}
+
 	_, err := c.caller.Call(ctx, "tx_search", params, result)
 	if err != nil {
 		return nil, err
 	}
+
+	return result, nil
+}
+
+func (c *baseRPCClient) BlockSearch(
+	ctx context.Context,
+	query string,
+	prove bool,
+	page,
+	perPage *int,
+	orderBy string,
+) (*ctypes.ResultBlockSearch, error) {
+
+	result := new(ctypes.ResultBlockSearch)
+	params := map[string]interface{}{
+		"query":    query,
+		"prove":    prove,
```

Comment:
Are these block results provable at all? 

Unless it really works, maybe remove the prove flag to not pretend to have an API that is not supported underneath. You can always add additional options when it is supported (that is a non-breaking api change)

Reply:
They're not. Good catch, I missed this!